src/components/NamespaceSelector/NamespaceSelectorTs.ts

```diff
@@ -49,12 +49,26 @@ export class NamespaceSelectorTs extends Vue {
    */
   @Prop({ default: null }) value: string
 
+  /**
+   * isLink
+   * True represents namespaces link, or false represents creating subnamespaces
+   * @type {string}
+   */
+  @Prop({ default: false }) isLink: boolean
+
   /**
    * Namespaces names
    * @type {[h: string]: string}
    */
   public namespaces: NamespaceModel[]
 
+  /**
+   * filter Namespaces
+   */
+  get getNamespaces() {
+    return this.namespaces.filter((n) => (this.isLink ? n.aliasType.valueOf() === 0 : n.depth.valueOf() !== 3))
+  }
+
```

Comment:
please also add `get unlinkableNamespaces` and should always returns only **namespaces that are linked to an alias**.

Reply:
please also add `get allNamespaces` and should return **all namespaces**.

Reply:
and then finally also add `get filteredNamespaces` which, depending on `disableLinked` and `disableUnlinked` should decide which list has to be displayed.